syntax = "proto3";

message BfPipelineConfig {
  message Profile {
    string profile_name = 1;         // name for this profile
    bytes context = 2;               // context.json file
    bytes binary = 3;                // tofino.bin file
    repeated uint32 pipe_scope = 4;  // list of hardware pipes for this profile
  }
  string p4_name = 1;             // name for this P4 program
  bytes bfruntime_info = 2;       // bfrt.json file
  repeated Profile profiles = 3;  // list of profiles for this program
}